Source/Core/Core/PowerPC/MMU.cpp

```diff
@@ -1177,27 +1177,29 @@ static void UpdateBATs(BatTable& bat_table, u32 base_spr)
       {
         // This bit is a little weird: if BRPN & j != 0, we end up with
         // a strange mapping. Need to check on hardware.
-        u32 address = (batl.BRPN | j) << BAT_INDEX_SHIFT;
+        u32 physical_address = (batl.BRPN | j) << BAT_INDEX_SHIFT;
+        u32 virtual_address = (batu.BEPI | j) << BAT_INDEX_SHIFT;
 
         // The bottom bit is whether the translation is valid; the second
         // bit from the bottom is whether we can use the fastmem arena.
-        u32 valid_bit = 0x1;
-        if (Memory::m_pFakeVMEM && ((address & 0xFE000000) == 0x7E000000))
-          valid_bit = 0x3;
-        else if (address < Memory::REALRAM_SIZE)
-          valid_bit = 0x3;
-        else if (Memory::m_pEXRAM && (address >> 28) == 0x1 &&
-                 (address & 0x0FFFFFFF) < Memory::EXRAM_SIZE)
-          valid_bit = 0x3;
-        else if ((address >> 28) == 0xE && (address < (0xE0000000 + Memory::L1_CACHE_SIZE)))
-          valid_bit = 0x3;
+        u32 valid_bit = BAT_MAPPED_BIT;
+        if (Memory::m_pFakeVMEM && ((physical_address & 0xFE000000) == 0x7E000000))
```

Comment:
TBH, I don't mind keeping parentheses for readability sake. Especially for the others cases I pointed out where it does help readibility. However, in that case it makes it harder to read IMO. If I remove the left condition, it gives:
```C++
if (((physical_address & 0xFE000000) == 0x7E000000))
```
Rather than having:
```C++
if (Memory::m_pFakeVMEM && (physical_address & 0xFE000000) == 0x7E000000)
```